CsvTextFieldParser/CsvTextFieldFormatter.cs

```diff
@@ -0,0 +1,293 @@
+            : this(new StreamWriter(path, append)) { }
+
+        /// <summary>
+        /// Constructs a formatter from the specified output file path with the specified encoding.
+        /// If the file exists, it can be either overwritten or appended to.
+        /// If the file does not exist, this constructor creates a new file.
+        /// </summary>
+        public CsvTextFieldFormatter(string path, bool append, Encoding encoding)
+            : this(new StreamWriter(path, append, encoding)) { }
+
+        /// <summary>
+        /// Constructs a formatter from the specified input text writer.
+        /// </summary>
+        public CsvTextFieldFormatter(TextWriter writer)
+        {
+            if (writer == null) throw new ArgumentNullException(nameof(writer));
+            this.writer = writer;
+        }
+
+        /// <summary>
+        /// Writes the fields to a new line.
+        /// </summary>
+        /// <param name="fields">An array of strings that contains field values for the current line.</param>
+        public void WriteFields(IEnumerable<string> fields)
+        {
+            if (writer == null) throw new ObjectDisposedException(nameof(writer));
+            if (fields == null) throw new ArgumentNullException(nameof(fields));
+
+            using (var enumerator = fields.GetEnumerator())
+            {
+                if (enumerator.MoveNext())
+                {
+                    WriteField(enumerator.Current);
+                    while (enumerator.MoveNext())
+                    {
+                        writer.Write(delimiterChar);
+                        WriteField(enumerator.Current);
+                    }
+                }
+            }
+            WriteEol();
+        }
+
+        private void WriteField(string field)
+        {
+            var isQuoteNecessary = field.IndexOfAny(SpecialChars) >= 0;
+            if (isQuoteNecessary)
+            {
+                writer.Write(quoteChar);
+                foreach (var ch in field)
+                {
+                    if (ch == quoteChar)
+                    {
+                        writer.Write(quoteEscapeChar);
+                    }
+                    writer.Write(ch);
+                }
+                writer.Write(quoteChar);
+            }
+            else
+            {
+                writer.Write(field);
+            }
+        }
+
+        private void WriteEol()
+        {
+            if (eol != null)
+            {
+                writer.Write(eol);
+            }
+            else
+            {
+                writer.WriteLine();
+            }
+        }
+
+        private char[] SpecialChars
+        {
+            get
+            {
+                var specialChars = specialCharsCache;
+                if (specialChars == null)
+                {
+                    specialChars = new SortedSet<char> { delimiterChar, quoteChar, quoteEscapeChar, '\n', '\r' }.ToArray();

```

Comment:
I wonder, why go via SortedSet instead of simply new char[]{  delimiterChar, quoteChar, quoteEscapeChar, '\n', '\r'}